hip_data_tools/etl/google_sheet_to_athena.py

```diff
@@ -1,242 +1,104 @@
-            self.__validate_field_names_and_types_count(field_names, field_types)
-            self.__validate_field_names(field_names)
-        else:
-            for field in self.settings.fields:
-                field_name_type = field.split(':')
-                field_name = field_name_type[0]
-                field_names.append(field_name)
-                field_types.append(field_name_type[1])
-                self.__validate_field_names([field_name])
+        s3_key_with_partition = self.settings.target_s3_dir
+        if self.settings.manual_partition_key_value is not None:
+            column_name = self.settings.manual_partition_key_value["column"]
+            column_value = self.settings.manual_partition_key_value["value"]
+            partition_path = f"/{column_name}={column_value}"
+            s3_key_with_partition += partition_path
+        self.write_sheet_data_to_s3(s3_key_with_partition)
 
-        values_matrix = sheet_util \
-            .get_value_matrix(workbook_name=self.settings.workbook_name,
-                              sheet_name=self.settings.sheet_name,
-                              row_range=self.settings.row_range,
-                              skip_top_rows_count=self.settings.skip_top_rows_count)
-        log.info("The value matrix:\n %s", values_matrix)
-        table_settings = self._get_table_settings(field_names, field_types)
-        athena_util.create_table(table_settings)
-        insert_query = self._get_the_insert_query(values_matrix=values_matrix, types=field_types)
-        log.info("The insert query:\n %s", insert_query)
-        athena_util.run_query(query_string=insert_query)
-
-    @staticmethod
-    def __validate_field_names_and_types_count(field_names, field_types):
-        if len(field_names) != len(field_types):
-            log.error("Number of field names and number of field types are not matching")
-            raise Exception("Field names and types are not matching")
-
-    @staticmethod
-    def __validate_field_names(field_names):
-        for field_name in field_names:
-            # check for strings which only contains letters, numbers and underscores
-            if not re.match("^[A-Za-z0-9_]+$", field_name):
-                log.error("Unsupported field name: %s", field_name)
-                raise Exception("Unsupported field name")
+        athena_util = self._get_athena_util()
+        if self.settings.target_table_ddl_progress:
+            athena_util.drop_table(self.settings.target_table_name)
+
+        athena_util.create_table(table_settings=get_table_settings_for_sheets_table(
+            dataframe=self._get_sheet_dataframe(),
+            partitions=self.settings.manual_partition_key_value,
+            table=self.settings.target_table_name,
+            s3_bucket=self.settings.target_s3_bucket,
+            s3_dir=self.settings.target_s3_dir))
+        athena_util.repair_table_partitions(table=self.settings.target_table_name)
```

Comment:
this should happen only when `self.settings.target_table_ddl_progress = true`
In other cases you should just add partition like so:
`athena_util.add_partitions(table=self.settings.target_table_name, partition_keys=[self.settings.manual_partition_key_value...], partition_values=[...])`

Reply:
Fixed